rcl/src/rcl/arguments.c

```diff
@@ -139,6 +144,28 @@ rcl_arguments_get_param_overrides(
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_arguments_get_log_level(
+  const rcl_arguments_t * arguments,
+  rcl_log_level_t ** log_level)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(arguments, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(arguments->impl, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(log_level, RCL_RET_INVALID_ARGUMENT);
+
+  if (NULL != *log_level) {
+    RCL_SET_ERROR_MSG("Output log level pointer is not null. May leak memory.");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  if (NULL != arguments->impl->log_level) {
+    *log_level = rcl_log_level_copy(arguments->impl->log_level);
+    if (NULL == *log_level) {
+      return RCL_RET_BAD_ALLOC;
+    }
+  }
```

Comment:
So if `arguments->impl->log_level` is NULL, then we'll leave `*log_level` as NULL.  Is that expected?  What happens when we go to free?

Reply:
This seems to be solved in line 153 now